control-plane/csi-driver/src/bin/controller/controller.rs

```diff
@@ -605,7 +605,17 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
             return Ok(Response::new(ControllerUnpublishVolumeResponse {}));
         }
 
-        // Do forced volume upublish as Kubernetes already detached the volume.
+        // no-op in case the node where unpublish is coming from isn't the current frontend node.
+        // This might happen in case of out of order/delayed unpublish call after application is
+        // is already published elsewhere.
+        if let Some(tgt) = volume.spec.target {
```

Comment:
I think this needs to be done at the core-agent unpublish, otherwise it's racy.
We already do similar thing for the republish: https://github.com/openebs/mayastor-control-plane/blob/dd21e4e276d2d8701d3935c2299219a234ccd955/control-plane/agents/src/bin/core/volume/operations.rs#L467

Reply:
Because we have resource locks in core-agent? Then I'd say yes that'd be better.

Reply:
I think both the places?

Reply:
> Because we have resource locks in core-agent? Then I'd say yes that'd be better.

Yes

> I think both the places?

Just core-agent, otherwise here it's still racy

Reply:
Made the changes now.